src/Analyzers/Core/CodeFixes/RemoveRedundantEquality/RemoveRedundantEqualityCodeFixProvider.cs

```diff
@@ -0,0 +1,99 @@
+
+namespace Microsoft.CodeAnalysis.RemoveRedundantEquality
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    internal sealed class RemoveRedundantEqualityCodeFixProvider
+        : SyntaxEditorBasedCodeFixProvider
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public RemoveRedundantEqualityCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(IDEDiagnosticIds.RemoveRedundantEqualityDiagnosticId);
+
+        internal override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                context.RegisterCodeFix(new MyCodeAction(
+                    AnalyzersResources.Remove_redundant_equality,
+                    c => FixAsync(context.Document, diagnostic, c)),
+                    diagnostic);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        protected override async Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = root.FindNode(diagnostic.AdditionalLocations[0].SourceSpan, getInnermostNodeForTie: true);
+
+                editor.ReplaceNode(node, (n, _) =>
+                {
+                    if (!syntaxFacts.IsBinaryExpression(n))
+                    {
+                        // This should happen only in error cases.
+                        return n;
+                    }
+
+                    syntaxFacts.GetPartsOfBinaryExpression(n, out var left, out var right);
+                    if (diagnostic.Properties[RedundantEqualityConstants.RedundantSide] == RedundantEqualityConstants.Right)
+                    {
+                        return WithElasticTrailingTrivia(left);
+                    }
+                    else if (diagnostic.Properties[RedundantEqualityConstants.RedundantSide] == RedundantEqualityConstants.Left)
+                    {
+                        // Elastic trivia is not needed here.
+                        // LeftExpression                 ==                     RightExpression
+                        // |This is the left operand span|This is the token span|This is the right operand span
```

Comment:
why not just always add the elastic on whichever part you're keeping.  would make this a lot simpler :)

Reply:
Just wanted to avoid unnecessary call. But if you want to always call AsElastic I'm okay with that.

Reply:
yup, let's just do that.  it will likely be more ideal as well in terms of interesting wrapping/formatting cases.

           -- Cyrus